Tests/BrowserServicesKitTests/ContentBlocker/SurrogatesUserScriptTests.swift

```diff
@@ -215,16 +281,22 @@ class SurrogatesUserScriptsTests: XCTestCase {
                                                                   trackerAllowlist: [:],
                                                                   contentBlockingEnabled: true,
                                                                   exceptions: [])
+        userScriptDelegateMock.shouldProcessCTLTrackers = true
+
         let websiteURL = URL(string: "test://example.com")!
 
         let websiteLoaded = self.expectation(description: "Website Loaded")
         let surrogateValidated = self.expectation(description: "Validated surrogate injection")
 
         navigationDelegateMock.onDidFinishNavigation = {
-            websiteLoaded.fulfill()
 
-            XCTAssertEqual(self.userScriptDelegateMock.detectedSurrogates.count, 1)
-            XCTAssertEqual(self.userScriptDelegateMock.detectedSurrogates.first?.url, self.surrogateScriptURL.absoluteString)
+            Task {
+                await self.sleepFor(secs: self.fulfillWait)
```

Comment:
Yea this is not good - any timing change could result in this test being flaky and randomly failing. Have you consider putting the code below at that end of the `self.webView?.evaluateJavaScript` completion?

Reply:
@bwaresiak if you think that's a safer approach, that's fine by me.  Either way I think we have a race condition, so the hope is the JS eval callback will complete after the ctlEnabled message completes the roundtrip.  I made this change and tested it, seems to work so I removed the timeouts.